silence/silence.go

```diff
@@ -447,33 +501,14 @@ func QTimeRange(start, end time.Time) QueryParam {
 }
 
 // QMatches returns silences that match the given label set.
-func QMatches(set map[string]string) QueryParam {
+func QMatches(set model.LabelSet) QueryParam {
```

Comment:
we either use the `model.LabelSet` here or change all the types in `types/match.go`, `dispatch/route.go`, `inhibit/inhibit.go` .. having `map[string]string` was much easier before as we duplicated all the matching functionality... which trade-off should we take?